src/controllers/dynakube/connectioninfo/communication_hosts.go

```diff
@@ -16,3 +21,59 @@ func GetOneAgentCommunicationHosts(dynakube *dynatracev1beta1.DynaKube) []dtclie
 	}
 	return communicationHosts
 }
+
+func GetActiveGateEndpointsAsCommunicationHosts(dynakube *dynatracev1beta1.DynaKube) []dtclient.CommunicationHost {
+	activegateEndpointsString := dynakube.Status.ActiveGate.ConnectionInfoStatus.Endpoints
+	if activegateEndpointsString == "" {
+		return []dtclient.CommunicationHost{}
+	}
+	return parseCommunicationHostFromActiveGateEndpoints(activegateEndpointsString)
+}
+
+func parseCommunicationHostFromActiveGateEndpoints(activegateEndpointsString string) []dtclient.CommunicationHost {
+	endpointStrings := strings.Split(activegateEndpointsString, ";")
+	endpointCommunicationHosts := make([]dtclient.CommunicationHost, 0)
+	for _, endpointString := range endpointStrings {
+		endpoint, err := parseEndpointToCommunicationHost(endpointString)
+		if err == nil {
+			endpointCommunicationHosts = append(endpointCommunicationHosts, endpoint)
+		}
+	}
```

Comment:
be aware that the endpoints might have duplicates in them, which is not allowed in `VirtualServices`
here is the recent bugfix for it: https://github.com/Dynatrace/dynatrace-operator/pull/2167
- that fix was aimed at the OneAgent Status as we store the OneAgent `CommunicationHosts` and just use them in the istio reconciler
- we don't do such a thing for the ActiveGate Status, so you have to do it here, and in the future we can consider to also store the `CommunicationHosts` of the ActiveGate in the Status

Reply:
done